src/main/java/org/folio/circulation/services/EventPublisher.java

```diff
@@ -336,12 +344,23 @@ public EventPublisher(PubSubPublishingService pubSubPublishingService) {
     write(eventJson, PAYLOAD.value(), context);
 
     return pubSubPublishingService.publishEvent(LOG_RECORD.name(), eventJson.encode())
-      .thenApply(r -> succeeded(null));
+      .handle((result, error) -> {
+        if (error != null) {
+          return failedDueToServerError(error.getMessage());
+        }
+        return succeeded(null);
+      });
```

Comment:
you can use the method below with null argument

Reply:
the method return value is not compatible, it requires `Result<Void>`

Reply:
Why can't we pass null as a loan to private Result<Loan> handlePublishEventError(Throwable error, Loan loan) {?

Reply:
because that private method returns `Result<Loan>`, the case you want me to use that method requires `Result<Void>`

Reply:
resolved, made that private method more common